sheepdog/transactions/upload/sub_entities.py

```diff
@@ -330,13 +330,24 @@ def _update_index(self):
         # If indexd record is not replaceable, update existing document
         document = self.get_indexed_document()
         if not self._is_replaceable:
-
             if self.urls:
                 urls_to_add = [url for url in self.urls
                                if url not in document.urls]
                 if urls_to_add:
                     document.urls.extend(urls_to_add)
-                    document.patch()
+            
+            document.hashes['md5'] = self.doc['md5sum']
+            document.file_size = self.doc['file_size']
+            document.file_name = self.doc['file_name']
+            if self.doc.get('metadata'):
+                document.metadata.update(self.doc.get('metadata'))
+            # document.acl = self.node.acl or self.get_file_acl()
+            document.urls_metadata = {
```

Comment:
this will reset the state and type of all urls in the document. Not sure this is what you want to do, maybe only reset if there is a new url

Reply:
This is a bit counter intuitive, but this is the current behavior. At this point there should be just one cleversafe url. The cleversafe url state will go back to register, as a result user needs to reupload file again, no matter what field was updated.